Tools/WrapperGenerator/CSharpWrapperGenerator.py

```diff
@@ -70,48 +83,49 @@ def __translate_type(self, obj: TypeMetadata) -> None:
         t = obj.name
 
         if not hasattr(obj, "is_void_pointer"):
-            setattr(obj, "is_void_pointer", False)
-
-        if t == "int8_t":
-            obj.name = "sbyte"
-        elif t == "uint8_t" or t == "unsigned char":
-            obj.name = "byte"
-        elif t == "int16_t":
-            obj.name = "short"
-        elif t == "uint16_t":
-            obj.name = "ushort"
-        elif t == "int32_t" or t == "long":
-            obj.name = "int"
-        elif t == "uint32_t" or t == "unsigned int" or t == "unsigned long":
-            obj.name = "uint"
-        elif t == "int64_t" or t == "long long" or t == "long int":
-            obj.name = "long"
-        elif t == "uint64_t" or t == "unsigned long long" or t == "unsigned long int":
-            obj.name = "ulong"
-        elif t == "size_t":
-            # Assume 64-bit only for now
-            obj.name = "ulong"
-        elif t == "String":
-            obj.name = "string"
-            obj.namespace = None
-            setattr(obj, "translated_namespace", None)
-            obj.is_pointer = False
-            obj.is_reference = False
-            obj.is_pointer_or_reference = False
-        elif (t == "void" or t == "char") and obj.is_pointer:
-            obj.name = "IntPtr"
-            obj.is_pointer = False
-            obj.is_reference = False
-            obj.is_pointer_or_reference = False
-            setattr(obj, "is_void_pointer", True)
-            setattr(obj, "translated_namespace", None)
+            obj.is_void_pointer = False
+
+        match t:
```

Comment:
is it worth considering introducing a default case with suitable error handling here, such as `case _: "Log Some Additional Information"`

Reply:
The way this function works, it'll translate types if it knows about them - otherwise it'll leave them as-is. So the default is to just leave the type name alone (with the namespace already translated by `__translate_namespace()`).

(This function would be a great candidate for unit tests!)

Reply:
Ah that is great, thank you for the clarification, and I do agree this would be a prime candidate for some unit testing. Happy to resolve this conversation.